src/ember-app.js

```diff
@@ -287,77 +111,203 @@ class EmberApp {
+  let vendorFile = fs.readFileSync(vendorFilePath, 'utf8');
+
+  debug("evaluating app; app=%s; vendor=%s", appFilePath, vendorFilePath);
+
+  sandbox.eval(vendorFile, vendorFilePath);
+  debug("vendor file evaluated");
+
+  sandbox.eval(appFile, appFilePath);
+  debug("app file evaluated");
+}
+
+/**
+ * Creates a destroy function to destroy the created `ApplicationInstance`.
+ *
+ * @param {Object} The Result object that contains the application instance to be destroyed.
+ * @param {Integer} How long to wait before manually destroying the application instance. Ignored if its value is 0.
+ *
+ * @returns {Function} The destructor function.
+ */
+function destroyAppInstance(instance, result, timeout) {
+  if (!timeout) return () => {
+    debug('destroying application instance');
+    instance.destroy();
+  };
+
+  // start a timer to destroy the appInstance forcefully in the given ms.
+  // This is a failure mechanism so that node process doesn't get wedged if the `visit` never completes.
+  let destructionTimer = setTimeout(function() {
+    debug('destroying application instance because of timeout');
+    instance.destroy();
+    result.error = new Error(`App instance was forcefully destroyed in ${timeout}ms`);
+  }, timeout);
+
+  return () => {
+    debug('destroying application instance');
+    clearTimeout(destructionTimer);
+    instance.destroy();
+  };
+}
+
+/**
+  * @private
+  *
+  * Builds and initializes a new sandbox to run the Ember application in.
+  *
+  * @param {string} distPath path to the built Ember app to load
+  * @param {Sandbox} [sandboxClass=VMSandbox] sandbox class to use
+  * @param {Object} [sandboxGlobals={}] any additional variables to expose in the sandbox or override existing in the sandbox
+  */
+function buildSandbox(distPath, sandboxClass, sandboxGlobals, moduleWhitelist, appConfig) {
+  let Sandbox = sandboxClass || require('./vm-sandbox');
+  let sandboxRequire = buildWhitelistedRequire(moduleWhitelist, distPath);
+  let config = appConfig;
+
+  // add any additional user provided variables or override the default globals in the sandbox
+  let globals = {
+    najax: najax,
+    FastBoot: {
+      require: sandboxRequire,
+      config: () => ({ default: config })
+    }
+  };
+
+  for (let key in sandboxGlobals) {
+    if (sandboxGlobals.hasOwnProperty(key)) {
+      globals[key] = sandboxGlobals[key];
     }
+  }
 
-    let manifest;
-    let pkg;
+  return new Sandbox({ globals });
+}
 
-    try {
-      pkg = JSON.parse(file);
-      manifest = pkg.fastboot.manifest;
-    } catch (e) {
-      throw new Error(`${pkgPath} was malformed or did not contain a manifest. Ensure that you have a compatible version of ember-cli-fastboot.`);
+/**
+  * The Ember app runs inside a sandbox that doesn't have access to the normal
+  * Node.js environment, including the `require` function. Instead, we provide
+  * our own `require` method that only allows whitelisted packages to be
+  * requested.
```

Comment:
This comment seems a bit odd when reading. Can your re-word it to explain the usecase?